tests/models/multiple_choice_qa/multiple_choice_bidaf_test.py

```diff
@@ -0,0 +1,38 @@
+# pylint: disable=no-self-use,invalid-name
+from keras import backend as K
+
+from deep_qa.models.multiple_choice_qa.multiple_choice_bidaf import MultipleChoiceBidaf
+from deep_qa.models.reading_comprehension.bidirectional_attention import BidirectionalAttentionFlow
+from ...common.test_case import DeepQaTestCase
+from ...common.test_markers import requires_tensorflow
+
+
+class TestMultipleChoiceBidaf(DeepQaTestCase):
+    # Theano has some problems with saving the loaded model in h5, because of how it names weights
+    # (pretty sure this arises because we're using two different BiDAF-based submodels in our
+    # model).  We'll punt on fixing that for now; the fix would likely be a pretty major hack.
+    # Theano can _train_ the model alright, but it can't _save_ it, or (presumably) load it.
+    @requires_tensorflow
+    def test_trains_and_loads_correctly(self):
+        self.write_span_prediction_files()
+        args = {
+                'model_serialization_prefix': self.TEST_DIR + "_bidaf",
+                'embedding_size': 4,
+                'save_models': True,
+                'tokenizer': {'type': 'words and characters'},
+                'show_summary_with_masking_info': True,
+                }
+        bidaf_model = self.get_model(BidirectionalAttentionFlow, args)
+        bidaf_model.train()
+
+        bidaf_model_params = self.get_model_params(BidirectionalAttentionFlow, args)
+        args = {
+                'bidaf_params': bidaf_model_params,
+                'train_bidaf': False,
+                'similarity_function': {'type': 'linear', 'combination': 'x,y'},
+                }
+        self.write_who_did_what_files()
+        model, _ = self.ensure_model_trains_and_loads(MultipleChoiceBidaf, args)
+        # All of the params come from the linear similarity function in the attention layer,
+        # because we set `train_bidaf` to `False`.
+        assert sum([K.count_params(p) for p in model.model.trainable_weights]) == 41
```

Comment:
I looked at the output of the model summary...  =).  And then I checked that the number made sense.  It's a linear function, with a number of parameters equal to `|x| + |y| + 1` (`+ 1` because of the bias).  `|x|` here is `32`, because that's what you end up with in BiDAF, and `|y|` is `8`, because it's a bidirectional encoder with embedding size 4, I believe...